### PR TITLE
Setup Travis CI

### LINKED ISSUE #15: Setup Travis CI
Although this repository doesn't have tests, it would be still great if we use Travis CI to check whether this project is compilable without errors and warnings.

### PR DESCRIPTION
**Pull request by jslee02**
Resolves #15.

* This change only includes checking if the code can be built without compilation errors. We could consider adding unit tests if available in the future.
* Test on Xenial because [`rewd_controllers` requires Kinetic or higher](https://github.com/personalrobotics/rewd_controllers/blob/master/CMakeLists.txt#L23-L25)
* Remove dependency of `pr_hardware_interface` by copying necessary files from the repo. This is necessary because a public repo `rewd_controllers` cannot be dependent on a private repo `pr_hardware_interface`.

### CODE REVIEW

File: include/rewd_controllers/TriggerableInterface.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#ifndef REWD_CONTROLLERS_TRIGGERABLE_INTERFACE_HPP_
+#define REWD_CONTROLLERS_TRIGGERABLE_INTERFACE_HPP_
+
+#include <string>
+#include <hardware_interface/hardware_interface.h>
+#include <hardware_interface/internal/hardware_resource_manager.h>
+#include <pr_hardware_interfaces/TriggerState.h>
```
Comment (brianhou) [inline include/rewd_controllers/TriggerableInterface.hpp]: Should `TriggerState.h` also be changed?

**Comment by brianhou** [comment]
> Remove dependency of `pr_hardware_interface` by copying necessary files from the repo. This is necessary because a public repo `rewd_controllers` cannot be dependent on a private repo `pr_hardware_interface`.

It seems like it'd be better if we could either make `pr_hardware_interfaces` public, or migrate the relevant parts out into a public repo.

**Comment by jslee02** [comment]
> It seems like it'd be better if we could either make pr_hardware_interfaces public, or migrate the relevant parts out into a public repo.

I'm open to this changes. AFAIK, there is some license issue in making `pr_hardware_interfaces` public. @gilwoolee could you remind me what was the issue here and which option sounds better to you? :grin:

File: include/rewd_controllers/MoveUntilTouchController.hpp [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #include <hardware_interface/force_torque_sensor_interface.h>
 #include <hardware_interface/joint_command_interface.h>
 #include <pr_control_msgs/SetForceTorqueThresholdAction.h>
-#include <pr_hardware_interfaces/TriggerableInterface.h>
+#include <rewd_controllers/TriggerableInterface.hpp>
```
Comment (Tonkel) [inline include/rewd_controllers/MoveUntilTouchController.hpp]: In lines 24 and 76, there are two references to the pr_hardware_interfaces namespace left.

Comment (jslee02) [reply include/rewd_controllers/MoveUntilTouchController.hpp]: > not sure how to leave a comment on lines that are unchanged

In "File changed" tap, if you put your mouse cursor on slightly right to the line numbers, a blue button with plus icon will appear. You can leave a comment on lines by clicking the blue button.

**Comment by Tonkel** [comment]
The code is good, Travis seems good and the repo discussion is out of my area of expertise :)

File: include/rewd_controllers/MoveUntilTouchController.hpp (hunk 2 above)
Comment (Tonkel) [reply include/rewd_controllers/MoveUntilTouchController.hpp]: Thanks, but if I press the button on the bottom to show lines that are unchanged, I can hover over these lines, but the blue button doesn't appear for some reason...

Comment (jslee02) [reply include/rewd_controllers/MoveUntilTouchController.hpp]: Yeah, it seems GitHub only allows to add a comment for changed lines. :disappointed:

**Comment by jslee02** [comment]
Making `pr_hardware_interfaces` public out of https://github.com/personalrobotics/pr_ros_controllers sounds best option to me. Let me work on that if there is no objection.

**Comment by jslee02** [comment]
`pr_hardware_interfaces` is imported (https://github.com/personalrobotics/pr_hardware_interfaces/pull/1)

**Pull request merged**